api/src/main/java/org/apache/iceberg/expressions/Projections.java

```diff
@@ -205,20 +206,27 @@ private InclusiveProjection(PartitionSpec spec, boolean caseSensitive) {
     @Override
     @SuppressWarnings("unchecked")
     public <T> Expression predicate(BoundPredicate<T> pred) {
-      PartitionField part = spec().getFieldBySourceId(pred.ref().fieldId());
-      if (part == null) {
+      Collection<PartitionField> parts = spec().getFieldsBySourceId(pred.ref().fieldId());
+      if (parts == null) {
```

Comment:
Given how `org.apache.iceberg.PartitionSpec#fieldsBySourceId` field is implemented . Parts will never be null, so this if block seems redundant

Reply:
This could be null or empty because the predicate may be for a field that does not have partition columns. Say I have a table partitioned by `date(event_time), identity(log_level)` and I filter by `message = "some log msg"`. In that case, the reference is for the `message` field, which is not the source for any partition fields.

You might be talking about how the multimap will return an empty list, but I don't want to rely on that behavior if it changes in the future. When this was a singular lookup, it would return null, so I think it makes sense to still be defensive here.

Reply:
Hm, I think ur explanation makes sense. Thanks!